document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreQueryV1Test.java

```diff
@@ -1235,15 +1245,147 @@ public void testFindWithSortingAndPagination(String datastoreName) throws IOExce
+                    KeyExpression.of(new SingleValueKey(TENANT_ID, "7"))))
+            .build();
+
+    final Iterator<Document> resultDocs =
+        collection.find(Query.builder().setFilter(filter).build());
+    assertDocsAndSizeEqual(datastoreName, resultDocs, "query/key_filter_response.json", 1);
+  }
+
+  @ParameterizedTest
+  @MethodSource("databaseContextBoth")
+  public void testFindWithConflictingKeys(final String datastoreName) {
+    final Collection collection = getCollection(datastoreName);
+    final org.hypertrace.core.documentstore.query.Filter filter =
+        org.hypertrace.core.documentstore.query.Filter.builder()
+            .expression(
+                and(
+                    KeyExpression.of(new SingleValueKey(TENANT_ID, "7")),
+                    KeyExpression.of(new SingleValueKey(TENANT_ID, "8"))))
+            .build();
+
+    final Iterator<Document> resultDocs =
+        collection.find(Query.builder().setFilter(filter).build());
+    assertFalse(resultDocs.hasNext());
+  }
+
+  @ParameterizedTest
+  @MethodSource("databaseContextBoth")
+  public void testFindWithMultipleKeys(final String datastoreName) throws IOException {
+    final Collection collection = getCollection(datastoreName);
+    final org.hypertrace.core.documentstore.query.Filter filter =
+        org.hypertrace.core.documentstore.query.Filter.builder()
+            .expression(
+                or(
+                    KeyExpression.of(new SingleValueKey(TENANT_ID, "7")),
+                    KeyExpression.of(new SingleValueKey(TENANT_ID, "3"))))
+            .build();
+
+    final Iterator<Document> resultDocs =
+        collection.find(Query.builder().setFilter(filter).build());
+    assertDocsAndSizeEqualWithoutOrder(
+        datastoreName, resultDocs, 2, "query/key_filter_multiple_response.json");
+  }
+
+  @ParameterizedTest
+  @MethodSource("databaseContextBoth")
+  public void testFindWithNonExistingKeys(final String datastoreName) {
+    final Collection collection = getCollection(datastoreName);
+    final org.hypertrace.core.documentstore.query.Filter filter =
+        org.hypertrace.core.documentstore.query.Filter.builder()
+            .expression(KeyExpression.of(new SingleValueKey(TENANT_ID, "30")))
+            .build();
+
+    final Iterator<Document> resultDocs =
+        collection.find(Query.builder().setFilter(filter).build());
+    assertFalse(resultDocs.hasNext());
+  }
+
+  @ParameterizedTest
+  @MethodSource("databaseContextBoth")
+  public void testFindWithKeyAndMatchingRelationalFilter(final String datastoreName)
+      throws IOException {
+    final Collection collection = getCollection(datastoreName);
+    final org.hypertrace.core.documentstore.query.Filter filter =
+        org.hypertrace.core.documentstore.query.Filter.builder()
+            .expression(
+                and(
+                    KeyExpression.of(new SingleValueKey(TENANT_ID, "7")),
```

Comment:
**Just a note:** Being a document store, which allows storing data as (key, document), we can have query language for `key` and `document` parts separately. Here, we are taking a single Query language approach for both the fields - key and document. So the query can have both key and document-related expressions as shown here in the example.
This adds the functionality of selecting certain documents based on a set of IDs, and further filtering document values.

Reply:
Nit: Could be better if the test cases use multiple IDs as `OR` and further filter.

Reply:
> Here, we are taking a single Query language approach for both the fields - key and document. So the query can have both key and document-related expressions as shown here in the example.
This adds the functionality of selecting certain documents based on a set of IDs, and further filtering document values.

Yes. This is done intentionally so that we have can combine key and other fields in querying making the query language more expressive. For example, we have `delete(Key)` and `delete(Filter)`. Tomorrow if we have a use-case to delete a specific document only when a condition is satisfied (like delete the document with id 7 only if the last updated time is older than 30 days), then we can neither use `delete(Key)` nor `delete(Filter)`.

Making the `Key` a part of the `Filter` not only allows for such use-cases, but also helps in merging many methods.

Reply:
> Could be better if the test cases use multiple IDs as OR and further filter.

Sure, will add in a separate PR.